envelope.md

```diff
@@ -52,14 +52,52 @@ envelopes with individual signatures.
 }
 ```
 
+### Signature Extensions
+
+In addition to `keyid` and `sig`, a signature object may include an `extension`
+field to store ecosystem specific information pertaining to the signature.
+Extensions do not modify the signing workflow established in the
+[DSSE protocol](protocol.md). However, signers and verifiers may exchange
+information about the extensions used out-of-band.
+
+```json
+{
+  "payload": "<Base64(SERIALIZED_BODY)>",
+  "payloadType": "<PAYLOAD_TYPE>",
+  "signatures": [{
+    "keyid": "<KEYID>",
+    "sig": "<Base64(SIGNATURE)>",
+    "extension": {
+      "kind": "<EXTENSION_KIND>",
+      "ext": {...}
+    }
+  }]
+}
+```
+
+`extension.kind` is a string and the `EXTENSION_KIND` value must unambiguously
+identify the type of the extension. This in turn identifies the fields in the
+opaque object `ext`. Some well-known extension types MAY be registered and
+listed with their `ext` definition alongside the DSSE specification.
+
+Additionally, extensions must not contain any information such that the
+signature verification fails in its presence and passes in its absence.
```

Comment:
Is the converse possible?

Reply:
Do you mean verification passes with the ext but not without? Sure, if a client requires this information, say in the sigstore case. Does this need to be called out here explicitly?

Reply:
bump!